sec_certs/model/dependency_finder.py

```diff
@@ -0,0 +1,117 @@
+from typing import List, Set, Dict, Tuple
+
+
+class DependencyFinder:
```

Comment:
Overall very well written. It does not exactly adhere to sklearn `BaseEstimator` (nor it inherits from it), but I guess it does not make sense to do that in this case.

I was not examining the algorithmical correctness of the computation, just the overall look and feel.

Reply:
The "correctness" can be crosschecked against older version - happy to run it for you